proposed/vcpkg.md

```diff
@@ -0,0 +1,118 @@
+
+### Using vcpkg
+
+All repositories that use vcpkg must use the manifest mode of vcpkg to declare their dependencies. Classic mode is not supported.
+
+### Source-build partners and conditionally system dependencies
+
+For some dependencies, we may want to use a system-installed version of the dependency in source-build scenarios.
+For these cases, our source-build partners can remove the dependencies from the vcpkg manifest.
+Since the CMake integration with vcpkg uses the standard `find_library` and `find_package` CMake commands, the system dependencies will be found instead of the vcpkg dependencies with no change to the CMake source code.
+
+### Flowing native dependencies with binary caching
+
+vcpkg has a binary caching feature that can use NuGet.
+We should be able to utilize this feature for flowing native dependencies that are consumed by native code (such as `dotnet/llvm-project` assets) to avoid having to run the build through MSBuild to locate the dependencies or rebuilding the dependencies in each repository. This is not something that needs to be done immediately, but is something we can consider in the future instead of our existing package flow model.
+
+## Q & A
+
+- Why use vcpkg instead of another package manager?
+    - Vcpkg is already used by many teams at Microsoft, and is already used by many of the dependencies that we may want to consume.
+    - Vcpkg is cross-platform.
+    - Vcpkg allows CMake consumers to consume dependencies as if they were installed on the machine instead of requiring custom CMake logic to find dependencies. Supporting changing a package between a vcpkg and system dependency would require more work with other package managers.
+- What is vcpkg's minimum toolset requirements? Do these conflict with the requirements for .NET?
+    - Vcpkg requires CMake 3.14 or newer. We require 3.20 or newer, so we should be fine.
+    - Vcpkg requires a C++ compiler that supports C++17. Although we don't use C++17, all of the compilers we support do support C++17.
+- Which packages would be migrated to use vcpkg?
+    - In dotnet/runtime, we would migrate the following dependencies:
+        - libunwind
+          - Given that we already have to support a system-provided version of libunwind
+    - In dotnet/runtime, we could migrate the following dependencies:
+        - Rapidjson
+        - zlib
+        - zlib-intel
+        - brotli
+    - If we switch to zlib-ng from zlib and zlib-intel, we would consume zlib-ng through vcpkg.
+    - In dotnet/aspnetcore we would migrate the following dependencies if we were to support vcpkg + MSBuild:
+        - GoogleTest
+    - Some libraries that we're looking at productizing in the future have these dependencies, which we'd also use vcpkg for:
+        - GoogleTest
```

Comment:
GoogleTest and Google Benchmark are test-only dependencies. They do not need to participate in the official or offline builds, with all the requirements. I guess they are the best place to start.